data/trie/snapshotDb.go

```diff
@@ -29,7 +29,9 @@ func (s *snapshotDb) DecreaseNumReferences() {
 		return
 	}
 	if s.numReferences == 0 && s.shouldBeDisconnected {
-		disconnectSnapshot(s.DBWriteCacher)
+		if err := disconnectSnapshot(s.DBWriteCacher); err != nil {
```

Comment:
I think it will look better if you split it like this:

err := disconnectSnapshot(s.DBWriteCacher)
if err != nil .....
 
Also, we use the same pattern throughout the rest of the project.

Reply:
Done!